connector_prestashop/product_combination.py

```diff
@@ -0,0 +1,402 @@
+                        binder.to_odoo(x.get('id'), unwrap=True) for x in
+                        images]
+                else:
+                    img_ids = []
+                if img_ids:
+                    product_binder = self.binder_for(
+                        'prestashop.product.combination')
+                    product_product = product_binder.to_odoo(
+                        combination['id'], unwrap=True, browse=True)
+                    product_product.with_context(
+                        connector_no_export=True).write(
+                        {'image_ids': [(6, 0, img_ids)]})
+            except PrestaShopWebServiceError:
+                pass
+
+    def import_supplierinfo(self, erp_id):
+        ps_id = self._get_prestashop_data()['id']
+        filters = {
+            # 'filter[id_product]': ps_id,
+            'filter[id_product_attribute]': ps_id
+        }
+        import_batch(
+            self.session,
+            'prestashop.product.supplierinfo',
+            self.backend_record.id,
+            filters=filters
+        )
+        ps_product_template = erp_id
+        ps_supplierinfos = self.env['prestashop.product.supplierinfo']. \
+            search([('product_tmpl_id', '=', ps_product_template.id)])
+        for ps_supplierinfo in ps_supplierinfos:
+            try:
+                ps_supplierinfo.resync()
+            except PrestaShopWebServiceError:
+                ps_supplierinfo.odoo_id.unlink()
+
+
+@prestashop
+class ProductCombinationMapper(ImportMapper):
+    _model_name = 'prestashop.product.combination'
+
+    direct = [
+    ]
+
+    from_main = []
+
+    @mapping
+    def combination_default(self, record):
+        return {'default_on': bool(int(record['default_on'] or 0))}
+
+    @mapping
+    def product_tmpl_id(self, record):
+        template = self.main_template(record)
+        return {'product_tmpl_id': template.odoo_id.id}
+
+    @mapping
+    def from_main_template(self, record):
+        main_template = self.main_template(record)
+        result = {}
+        for attribute in self.from_main:
+            if attribute not in main_template:
+                continue
+            if hasattr(main_template[attribute], 'id'):
+                result[attribute] = main_template[attribute].id
+            elif type(main_template[attribute]) is browse_record_list:
+                ids = []
+                for element in main_template[attribute]:
+                    ids.append(element.id)
+                result[attribute] = [(6, 0, ids)]
+            else:
+                result[attribute] = main_template[attribute]
+        return result
+
+    def main_template(self, record):
+        if hasattr(self, '_main_template'):
+            return self._main_template
+        template_id = self.get_main_template_id(record)
+        self._main_template = self.env['prestashop.product.template'].browse(
+            template_id)
+        return self._main_template
```

Comment:
this is a dangerous thing, the same mapper can be used for several `MapRecord`, nothing should be stored on `self`. (not sure to see the benefits BTW)